backends/dpdk/dpdkArch.cpp

```diff
@@ -2917,16 +2917,66 @@ MoveNonHeaderFieldsToPseudoHeader::addAssignmentStmt(const IR::Expression *e) {
     } else {
         type = typeMap->getType(e, true)->to<IR::Type_Bits>();
     }
+    if (e->is<IR::Constant>() && (type->width_bits() > 64)) {
+        type = IR::Type_Bits::get(64);
```

Comment:
this looks strange to me

Reply:
Target supports constants only up-to 64-bits. Perhaps, adding a check and issuing error in case the constant value doesn't fit in 64-bit would be better .I will update the code for that.